crates/solidity/testing/snapshots/cst_output/ContractDefinition/unicode_in_doc_comments/generated/0.4.11-failure.yml

```diff
@@ -9,14 +9,19 @@ Source: >
 Errors: # 1 total
   - >
     Error: Expected ContractKeyword.
-       ╭─[crates/solidity/testing/snapshots/cst_output/ContractDefinition/unicode_in_doc_comments/input.sol:1:1]
+       ╭─[crates/solidity/testing/snapshots/cst_output/ContractDefinition/unicode_in_doc_comments/input.sol:4:1]
        │
-     1 │ ╭─▶ // ╒════════════════════════════════════════════════════════════════╕
-       ┆ ┆   
-     4 │ ├─▶ unexpected
-       │ │                
-       │ ╰──────────────── Error occurred here.
+     4 │ unexpected
```

Comment:
> Likely this deserves a changeset, since it's changing the way an UNRECOGNIZED node is return when it's in the top-most position.

You mean like this file? How about:

```
improved error recovery, where leading trivia are always parsed and included before an errornous terminal.
```

Reply:
IIUC, now any parse operation can return a `Nonterminal`, right?
I wonder if we should also change the type of `ParseOutput::tree()` from `Node` to `Nonterminal`. It seems more accurate since users always pass a `NonterminalKind` to `parse()`, so it makes sense to get back a `Nonterminal`. Thoughts?

Reply:
IIUC, now any parse operation can return a `Nonterminal`, right?
I wonder if we should also change the type of `ParseOutput::tree()` from `Node` to `Nonterminal`. Users always pass a `NonterminalKind` to `parse()` anyways. Thoughts?

Reply:
I think it makes sense. Shall we do that in a follow-up?

Reply:
Sounds great. I will create a separate issue.

Reply:
#1184

Reply:
Keeping this unresolved so that we add the changeset.